project_task_materials_stock/models/project.py

```diff
@@ -83,7 +83,13 @@ def write(self, vals):
                         task.material_ids.create_analytic_line()
                 else:
                     if task.unlink_stock_move():
-                        task.analytic_line_ids.unlink()
+                        if any(task.material_ids.mapped(
+                                'analytic_line_id.invoice_id')):
+                            raise exceptions.Warning(
+                                _("You can't move to a not consume stage if "
```

Comment:
"You cannot move back to a not consumed stage if analytic lines have been invoiced"